sagemaker-python-sdk/pytorch_horovod_mpi/pytorch_horovod_mpi.ipynb

```diff
@@ -0,0 +1,791 @@
+      "\n",
+      "2020-06-16 01:40:02 Completed - Training job completed\n",
+      "Training seconds: 181\n",
+      "Billable seconds: 181\n"
+     ]
+    }
+   ],
+   "source": [
+    "estimator.fit(inputs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hosting"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating an Endpoint"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After training, we need to use the `PyTorch` estimator object to create a `PyTorchModel` object and set a different `entry_point`, otherwise, the training script `train.py` will be used for inference. (Note that the new `entry_point` must be under the same `source_dir` as `train.py`). Then we use the `PyTorchModel` object to deploy a `PyTorchPredictor`. This creates a Sagemaker Endpoint -- a hosted prediction service that we can use to perform inference.\n",
+    "\n",
+    "An implementation of `model_fn` is required for inference script. We are going to use default implementations of `input_fn`, `predict_fn`, `output_fn` and `transform_fm` defined in [sagemaker-pytorch-containers](https://github.com/aws/sagemaker-pytorch-training-toolkit).\n",
+    "\n",
+    "Here's the inference script we're using:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pygmentize src/inference.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The arguments to the deploy function allow us to set the number and type of instances that will be used for the Endpoint. These do not need to be the same as the values we used for the training job. Here we will deploy the model to a single ml.c5.xlarge instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a PyTorchModel object with a different entry_point\n",
+    "model = estimator.create_model(entry_point=\"inference.py\", source_dir=\"src\")\n",
+    "# Deploy the model to a ml.c5.xlarge instance\n",
+    "predictor = model.deploy(initial_instance_count=1, instance_type=\"ml.c5.xlarge\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Classify Digits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now use this predictor to classify hand-written digits. We'll try to predict the images we displayed earlier."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "outputs = predictor.predict(images)\n",
+    "predictions = np.argmax(outputs, axis=1)\n",
+    "\n",
+    "print('Predictions:', ' '.join('%4s' % predictions[i] for i in range(4)))"
```

Comment:
it might be helpful to print out both the original labels or images and the predictions (so that the reader doesn't have to scroll back up to compare)